lib/helpers/versions.js

```diff
@@ -96,3 +107,26 @@ function isPastLatest(dep, version) {
   logger.warn(`No dist-tags.latest for ${dep.name}`);
   return false;
 }
+
+function computeRangedVersion(currentVersion, newVersion) {
+  const parsedRange = semverUtils.parseRange(currentVersion);
+  let rangedVersion = '';
+
+  // Check whether the current version is pinned, and if so, just return the
+  // new version, which, itself, should be a pinned version number.
+  if (parsedRange.length === 1 && parsedRange[0].operator === undefined) {
+    rangedVersion = newVersion;
+
+  // Check whether the existing version range is _simple_, meaning
+  // the range is a single operator, like `^`, `~`, `>=`, etc on a single operand, `1.0.0`.
```

Comment:
I believe [primitive operators](https://github.com/npm/node-semver#ranges), tilde and carrot.

However, I think _X-range_ may also be a single operator, but I'm not sure how to handle that one, so, yeah, I think locking down to `~` and `^` is probably the safe course of action for now.